### PR TITLE
Feat/Add Exit Game Functionality

### PR DESCRIPTION
**Pull request by giripatel**
Adds Exit Game Functionality

1) Implemented an "Exit Game" feature.
2) Users can now exit the game in the middle of playing.
3) The opponent will be considered as the winner with a 'PLAYER_EXIT' status.

[Screencast from 18-06-24 01:07:07 AM IST.webm](https://github.com/code100x/chess/assets/103880028/ae2dbe82-2f4e-4a50-bccf-a2640cade319)

@hkirat 
@siinghd  Please review and  suggest if any are needed

### CODE REVIEW

**Comment by siinghd** [comment]
@giripatel can we improve the little bit the ui? its weird the button down there

File: apps/frontend/src/screens/Game.tsx [hunk 1]
```diff
@@ -132,7 +130,8 @@ export const Game = () => {
 
         case GAME_ENDED:
           const wonBy = message.payload.status === 'COMPLETED' ? 
-            message.payload.result !== 'DRAW' ? 'CheckMate' : 'Draw' : 'Timeout';
+            message.payload.result !== 'DRAW' ? 'CheckMate' : 'Draw'
```
Comment (siinghd) [inline apps/frontend/src/screens/Game.tsx:133]: Can we improve this? it's too nasted... 
```

 let wonBy;
  switch (message.payload.status) {
    case 'COMPLETED':
      wonBy = message.payload.result !== 'DRAW' ? 'CheckMate' : 'Draw';
      break;
    case 'PLAYER_EXIT':
      wonBy = 'Player Exit';
      break;
    default:
      wonBy = 'Timeout';
  }

  setResult({
    result: message.payload.result,
    by: wonBy,
    blackPlayer: message.payload.blackPlayer,
    whitePlayer: message.payload.whitePlayer,
  });
  setStarted(false);
```

File: apps/frontend/src/screens/Game.tsx [hunk 2]
```diff
@@ -306,10 +315,12 @@ export const Game = () => {
               </div>
             </div>
             <div className="rounded-md pt-2 bg-bgAuxiliary3 flex-1 overflow-auto h-[95vh]">
-              {!started && (
-                <div className="pt-8 flex justify-center w-full">
+              {!started? (
```
Comment (siinghd) [inline apps/frontend/src/screens/Game.tsx:318]: What is this, not getting this condition.

why we show 
```
<div className="p-8 flex justify-center w-full">
                  <ExitGameModel onClick={() => handleExit()} />
                </div>
```
                
 if not started

**Review by siinghd — CHANGES_REQUESTED** [review]

**Comment by giripatel** [comment]
Hi @siinghd 

Thanks for the review.

Regarding UI changes.

The button has adjusted due to the responsiveness of the application. If I make the webpage a bit wider/full-screen, it will move to the right of the chessboard.

File: apps/frontend/src/screens/Game.tsx (hunk 2 above)
Comment (giripatel) [reply apps/frontend/src/screens/Game.tsx:318]: If the game is not started the Play button and Waiting loader will render, if it's started the Exit Button will render.
The <div> you have mentioned above is the component of Exit button.

Please let me know if it seems complex at first glance, i'll replace the ternary with if else.

**Comment by siinghd** [comment]
Thank you sir

**Comment by siinghd** [comment]
For the bounty please do wait, will submit the PR to kirat.

**Pull request merged**